micrometer-test/src/main/java/io/micrometer/core/tck/GaugeTest.java

```diff
@@ -71,15 +68,14 @@ default void mapSizeGauge(MeterRegistry registry) {
         Map<String, Integer> map = registry.gaugeMapSize("my.gauge", emptyList(), new HashMap<>());
         map.put("a", 1);
 
-        Gauge g = registry.find("my.gauge").gauge().orElse(null);
-        assertNotNull(g);
+        Gauge g = registry.mustFind("my.gauge").gauge();
         assertThat(g.value()).isEqualTo(1);
     }
 
     @Test
     @DisplayName("gauges that reference an object that is garbage collected report NaN")
     default void garbageCollectedSourceObject(MeterRegistry registry) {
         registry.gauge("my.gauge", emptyList(), (Map) null, Map::size);
-        assertThat(registry.find("my.gauge").value(Statistic.Value, 0).gauge()).isPresent();
+        assertThat(registry.mustFind("my.gauge").gauge().value()).matches(val -> Double.isNaN(val) || val == 0.0);
```

Comment:
This test's output changed in an interesting way. StatsD reports a `0` while Prometheus returns a `NaN` Since I wasn't familiar with the `RegistryCompatibilityTest` I wasn't sure how to override the StatsD `GaugeTest` from the others.